src/cast.c

```diff
@@ -100,8 +101,28 @@ Expression *implicitCastTo(Expression *e, Scope *sc, Type *t)
                 //printf("type %p ty %d deco %p\n", type, type->ty, type->deco);
                 //type = type->semantic(loc, sc);
                 //printf("type %s t %s\n", type->deco, t->deco);
-                e->error("cannot implicitly convert expression (%s) of type %s to %s",
-                    e->toChars(), e->type->toChars(), t->toChars());
+                char hint[128];
+                snprintf(hint, sizeof(hint), "");
+                if (t->isTypeDArray())
+                {
+                    if (TypeStruct* typeStruct = e->type->isTypeStruct())
+                    {
+                        if (typeStruct->sym->isInputRange())
+                        {
+                            snprintf(hint, sizeof(hint), ", suffix Range with std.array.array");
+                        }
```

Comment:
I suggest to rephrase that, as it can be easily misread (what is a "suffix range"?).  How about, "use std.array.array to force conversion".